proxy/router/src/lib.rs

```diff
@@ -98,16 +106,26 @@ where T: Recognize,
 impl<T> Router<T>
 where T: Recognize
 {
-    pub fn new(recognize: T) -> Self {
+    pub fn new(recognize: T, capacity: usize) -> Self {
         Router {
             inner: Arc::new(Mutex::new(Inner {
-                routes: Default::default(),
+                routes: IndexMap::default(),
```

Comment:
Since we know the capacity here, do you think its' worthwhile to go ahead and construct the map with `IndexMap::with_capacity`? That way, AFAICT, we shouldn't have to reallocate once the `Router` is built...

Reply:
Assuming the maximum capacity could be pretty big, we don't necessarily want to allocate memory to fit all of them if the it turns on there is never that many routes used.